src/util/string.h

```diff
@@ -24,3 +25,18 @@ class StringCollator {
   private:
     QCollator m_collator;
 };
+
+class LinkFormatter {
+  public:
+    // Helper to create html link strings to be used for ui files, mostly in
+    // Preferences dialogs.
+    QString coloredLinkString(
```

Comment:
Why is a class? This could be a free function (or a static one). Or are you planning to add new methods here later?

As an alternative, you could pass in the color once in the constructor instead of passing it in each function call.

Reply:
When we add the ability ro [style DlgTrackproperty](https://github.com/mixxxdj/mixxx/blob/f083c4fdec65c1adea31114c7563c3e989f4404f/src/library/dlgtrackinfo.cpp#L37) and other dialogs (like Preferences) a simple linkString() builder could be helpful. Didn't look into that, yet.

> This could be a free function (or a static one).
As an alternative, you could pass in the color once in the constructor

Can you point me some example? Am still a c++ entrepreneur, without basic knowledge appearantly and having hard times understanding standard coding terms:)

Reply:
A free function is just a function that does not belong to a class (i.e. without `class Foo {}` around it). E.g.:

```cpp
bool negate(bool value) {
    return !value;
}
```

And you call it like this:

```cpp
bool myval = negate(val);
```

This example only uses free functions: https://www.learn-cpp.org/en/Functions

A static method looks like this:

```cpp
class Foo {
  public: 
    static bool negate(bool value) {
        return !value;
    }
  private:
    // Make constructor private to prevent creating an object of this class. 
    Foo() {}
} 
```

And you call it like this:

```cpp
bool myval = Foo::negate(val);
```

https://www.learncpp.com/cpp-tutorial/static-member-functions/

Reply:
Thanks a lot for the explanation! I will look into that.

Regarding
> When we add the ability ro style DlgTrackproperty and other dialogs (like Preferences) a simple linkString() builder could be helpful. Didn't look into that, yet.

is it okay to merge as it is now?

Reply:
I learned: a free function in a header (string.h here) it needs to be `inline`
https://github.com/mixxxdj/mixxx/pull/3464/commits/850bbdf347c9b5ec72dd2539c25301f26abf5942

Reply:
Ready again..